core/base_service.go

```diff
@@ -409,11 +409,10 @@ func (service *BaseService) Request(req *http.Request, result interface{}) (deta
 	httpResponse, err = service.Client.Do(req)
 
 	if err != nil {
-		errMsg := err.Error()
 		if strings.Contains(err.Error(), SSL_CERTIFICATION_ERROR) {
-			errMsg = ERRORMSG_SSL_VERIFICATION_FAILED + "\n" + errMsg
+			err = fmt.Errorf(ERRORMSG_SSL_VERIFICATION_FAILED + "\n" + err.Error())
 		}
-		err = SDKErrorf(nil, errMsg, "no-connection-made", getComponentInfo())
+		err = SDKErrorf(err, "", "no-connection-made", getComponentInfo())
```

Comment:
@pyrooka Norbert - I want to publicly acknowledge that "you told me so" with respect to saving the previously returned errors in case we ever needed them! 😂 

Turns out, we do need them - so this PR is adding everything back in just like you recommended in my last PR 😛

Reply:
Some developers never learn... :joy:

Reply:
Haha, thanks for mentioning me, appreciate it! But I hope you didn't do this to protect yourself from future complications that might come up, so you can point at me and say "Hey, that was Norbert's idea, not mine..." 😂